kafka-rest/src/main/java/io/confluent/kafkarest/controllers/PartitionManagerImpl.java

```diff
@@ -79,7 +86,26 @@ public CompletableFuture<Optional<Partition>> getPartition(
             partitions -> findEntityByKey(partitions, Partition::getPartitionId, partitionId))
         .thenApply(partition -> partition.map(Collections::singletonList).orElse(emptyList()))
         .thenCompose(this::withOffsets)
-        .thenApply(partitions -> partitions.stream().findAny());
+        .thenApply(partitions -> partitions.stream().findAny())
+        .exceptionally(
+            exception -> {
+              if (exception != null) {
+                if (exception.getCause() instanceof UnknownTopicOrPartitionException) {
+                  String exceptionMessage =
+                      String.format(
+                          "This server does not host topic-partition %d for topic %s",
+                          partitionId, topicName);
+                  throw new UnknownTopicOrPartitionException(exceptionMessage, exception);
+                }
+              } else if (exception instanceof NotFoundException
+                  || exception.getCause() instanceof NotFoundException) {
+                throw new NotFoundException(exception.getCause());
+              } else if (exception instanceof RuntimeException
+                  || exception.getCause() instanceof RuntimeException) {
+                throw (RuntimeException) exception;
+              }
```

Comment:
Am I missing something or is this not meaningful code?
We'll get to the second and third clause (the `else if`s) only if the first clause is not satisfied, i.e. if `exception` is `null`. But then the second and third clause don't make any sense.
Were they supposed to be nested within the first null check?
- I think that null check is redundant as I don't think `exceptionally` will be called without a reference to the offending exception.